File: backend/app/schemas/agent.py
from typing import Optional, Dict, Any, List
from uuid import UUID
from datetime import datetime
from pydantic import Field
from app.schemas.base import BaseSchema

class AgentBase(BaseSchema):
    name: str = Field(..., max_length=100)
    avatar_url: Optional[str] = Field(None, max_length=500)
    provider: str = Field(..., max_length=50)
    model: str = Field(..., max_length=100)
    system_prompt: Optional[str] = None
    capabilities: List[str] = Field(default_factory=list)
    tool_config: Optional[Dict[str, Any]] = None

class AgentCreate(AgentBase):
    pass

class AgentUpdate(BaseSchema):
    name: Optional[str] = Field(None, max_length=100)
    avatar_url: Optional[str] = Field(None, max_length=500)
    provider: Optional[str] = Field(None, max_length=50)
    model: Optional[str] = Field(None, max_length=100)
    system_prompt: Optional[str] = None
    capabilities: Optional[List[str]] = None
    tool_config: Optional[Dict[str, Any]] = None
    is_active: Optional[bool] = None

class AgentResponse(AgentBase):
    id: UUID
    is_builtin: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime

class AgentVerifyRequest(BaseSchema):
    provider: str
    model: str
    system_prompt: Optional[str] = None

File: agenthub-web/src/types/agent.ts
export interface Agent {
  id: string;
  name: string;
  avatarUrl: string;
  provider: string;
  model: string;
  systemPrompt?: string;
  capabilities: string[];
  toolConfig: Record<string, unknown>;
  baseUrl: string;       // 模型基址（匹配后端 to_camel 输出）
  apiKey: string;         // API Key（匹配后端 to_camel 输出）
  isBuiltin: boolean;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateAgentParams {
  name: string;
  avatarUrl?: string;
  provider: string;
  model: string;
  systemPrompt?: string;
  capabilities?: string[];
  toolConfig?: Record<string, unknown>;
  baseUrl: string;       // 必填 — 模型基址
  apiKey: string;         // 必填 — API Key
}

export interface UpdateAgentParams {
  name?: string;
  avatarUrl?: string;
  provider?: string;
  model?: string;
  systemPrompt?: string;
  capabilities?: string[];
  toolConfig?: Record<string, unknown>;
  isActive?: boolean;
  baseUrl?: string;
  apiKey?: string;
}
